csbpg/db_dumps_assets/aws_fixes.sql

```diff
@@ -0,0 +1,15 @@
+-- Dumps don't include any passwords. Let's assign a test one to the admin user
+ALTER USER testuser WITH PASSWORD 'test-password';
+
+-- Dumps don't seem to be replicating original permissions for some objects.
+-- This is something PostgresDB should be able to do automatically.
+-- Until we find how to do this properly, lets hardcode some manual fixes.
+\connect postgres
+ALTER SCHEMA PUBLIC OWNER TO testuser;
+
+\connect template1
+ALTER SCHEMA PUBLIC OWNER TO testuser;
+
+\connect testdb
+ALTER SCHEMA PUBLIC OWNER TO testuser;
```

Comment:
question (no-blocking): what is these sentences for?

The database testdb is deleted in the dump `csbpg/db_dumps_assets/aws_pg14.sql`

Reply:
The database is deleted at the top of `csbpg/db_dumps_assets/aws_pg14.sql` but [is recreated in subsequent instructions](https://github.com/cloudfoundry/terraform-provider-csbpg/blob/7dc7a99c299e15ce6a1f3a5601df223390f3479c/testfixtures/db_dumps/aws_aurora_pg14.sql#L324).

This is so because [the dump was taken using the `--clean` flag](https://github.com/cloudfoundry/terraform-provider-csbpg/blob/7dc7a99c299e15ce6a1f3a5601df223390f3479c/testfixtures/db_dumps/aws_aurora_pg14.sql#L4).

I have experimented using different flags and discovered that using `--if-exists --clean` creates dumps which better resemble the original database.

Answering your question:
```
\connect testdb
ALTER SCHEMA PUBLIC OWNER TO testuser;
```

The instruction above fixes ownership of the PUBLIC schema after restoring the dump. For some reason, the dump doesn't include such instruction.

Without this instruction, the PUBLIC schema would be owned by `restoredump` user which makes absolutely no sense since there is no such user in GCP. That's why I think I might be missing some flags as there must be a way for dumps to restore the original ownership of schemas.

Reply:
Thank you very much @fnaranjo-vmw !!!